pkg/vault/dummy.go

```diff
@@ -18,12 +18,15 @@ package vault
 
 import (
 	"path"
-	"testing"
 
 	"github.com/hashicorp/vault/api"
 )
 
-func setupAppRole(t *testing.T, name, token, address string, secret bool) string {
+type testingLogger interface {
+	Fatalf(format string, args ...interface{})
+}
+
+func setupAppRole(t testingLogger, name, token, address string, secret bool) string {
```

Comment:
What about to return a tuple (string, error) and catch the errors in pkg/vault/vault_test.go
It would not be needed to use here testing or this interface .

```
func setupAppRole(t *testing.T, name, token, address string, secret bool) (string, error) {
        ...
	options = RequestOptions{Data: roleParams}
	_, err = v.Request("POST", roleURL, &options)
	if err != nil {
		return "", fmt.Errorf("couldn't create approle testrole: %s ", err)
	}
        ...
        return roleID, nil
}
```

Reply:
Umm, I like it, thanks!